snackpub/src/index.ts

```diff
@@ -0,0 +1,111 @@
+import { Server } from 'socket.io';
+
+import Env from './Env';
+import { maybeBindRedisAdapterAsync, maybeCloseRedisConnectionsAsync } from './RedisAdapter';
+import type {
+  ClientToServerEvents,
+  InterServerEvents,
+  ServerToClientEvents,
+  ShutdownSignal,
+  SocketData,
+} from './types';
+
+const debug = require('debug')('snackpub');
+
+function registerShutdownHandlers(server: Server) {
+  const shutdown = async (signal: ShutdownSignal) => {
+    console.log(
+      `Received ${signal}; the HTTP server is shutting down and draining existing connections`
+    );
+    await closeServerAsync(server);
+    await maybeCloseRedisConnectionsAsync();
+  };
+
+  // TODO: In Node 9, the signal is passed as the first argument to the listener
+  process.once('SIGHUP', () => shutdown('SIGHUP'));
+  process.once('SIGINT', () => shutdown('SIGINT'));
+  process.once('SIGTERM', () => shutdown('SIGTERM'));
+  // Nodemon sends SIGUSR2 when it restarts the process it's monitoring
+  process.once('SIGUSR2', () => shutdown('SIGUSR2'));
+}
+
+function closeServerAsync(server: Server): Promise<void> {
+  return new Promise((resolve) => {
+    server.close((error) => {
+      if (error) {
+        console.log('Failed to close server', error);
+      }
+      resolve();
+    });
+  });
+}
+
+async function runAsync() {
+  const io = new Server<ClientToServerEvents, ServerToClientEvents, InterServerEvents, SocketData>({
+    serveClient: false,
+  });
+
+  await maybeBindRedisAdapterAsync(io);
```

Comment:
Instead of `maybeBindRedisAdapterAsync`, I think it would be clearer to write:
```ts
if (Env.redisURL) {
  await bindRedisAdapterAsync(io);
}
```

This way it's easier to see when the Redis adapter is used, and also to implement logic like:

```ts
if (Env.nodeEnv === 'production') {
  assert(Env.redisURL, `Redis must be configured`);
  await bindRedisAdapterAsync(io);
}
```
^ although, I don't think that's quite the right logic because we may want to test Redis when NODE_ENV is development.

Reply:
yep the code later is changed as something like that: https://github.com/expo/snack/blob/fd213775d3b2b755e6b15e2e7ff5416d67f3c7fe/snackpub/src/index.ts#L73-L89

i'll add the code accordingly.

```ts
if (Env.nodeEnv === 'production') {
  assert(Env.redisURL, `Redis must be configured`);
}
```